### ISSUE TITLE
Small mistake in train script

### ISSUE THREAD

**Issue by sasanamari** [usnistgov/alignn#81]
Hello everyone :)

Thanks for great tool! I have found a small error in the "alignn/train.py" script while I was trying to use it for regression.

At line 866 we have:
`

    if config.n_early_stopping is not None:
        if classification:
            my_metrics = "accuracy"
        else:
            my_metrics = "mae"

        def default_score_fn(engine):
            score = engine.state.metrics[my_metrics]
            return score

        es_handler = EarlyStopping(
            patience=config.n_early_stopping,
            score_function=default_score_fn,
            trainer=trainer,
        )
`
The problem is that, as stated in the documentation of ignite, "An improvement is considered if the score is higher." In other words, EarlyStopping only checks for increases of a benefit/reward function to be maximized, like "accuracy". A decreasing cost function would trigger earlyStopping after "patience" number of epochs, even if the cost is still decreasing.

To monitor the decrease of a cost function, e.g. "mae", the default_score_fn() could return the negative value of the cost metric, as shown in the ignite documentation example here:
https://pytorch.org/ignite/generated/ignite.handlers.early_stopping.EarlyStopping.html

All the best

**Comment by bdecost** [comment]
Nice catch, thanks. I proposed a quick fix

I usually train to completion with a cosine schedule, so probably wouldn't have noticed this

**Comment by knc6** [comment]
Added in the main repo now:

https://github.com/usnistgov/alignn/blob/main/alignn/train.py#L877

**Comment by bdecost** [comment]
closing as resolved, thanks again for flagging this!
